[docs] Add eas build:resign command (#21184)
expo/expo: docs/pages/app-signing/app-credentials.mdx

### Clearing Credentials

When you use the `eas credentials` command to delete your credentials, this only removes those credentials from Expo's servers, **it does not delete the credentials from Apple's perspective**. This means that to fully delete your credentials (for example, if you want a new push notification key, however, you already have two), you'll need to do so from the [Apple Developer Console](https://developer.apple.com/account/resources/certificates/list).

> ### Resigning new credentials

Reviewer: ```suggestion
### Re-signing new credentials
```

> You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new adhoc provisioning profile. This helps reduce time when distributing internally, for example, when you have multiple test iOS devices or want to share a build with your team members.

Author: ```suggestion
You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally, for example, when you have multiple test iOS devices or want to share a build with your team members.
```

> You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally, for example, when you have multiple test iOS devices or want to share a build with your team members.

Reviewer: ```suggestion
You can use `eas build:resign` to codesign an existing **.ipa** for iOS to a new ad hoc provisioning profile. This helps reduce time when distributing internally — for example, if you want to add a new test device to an existing build, you can use this command to update the provisioning profile to include the device without rebuilding the entire app from scratch.
```

> Running the command will ask you to select a build that you want to resign. For example, running the command in an example project shows an available build:

Reviewer: ```suggestion
Running the command will ask you to select a build that you want to re-sign. For example, running the command in an example project shows an available build:
```

it's re-sign in a sentence but resign in a command. the convention is to remove - in command names

<ImageSpotlight
  alt="Running eas build:resign command shows a list of available builds in a project."
  src="/static/images/eas-build/eas-build-resign.png"
/>

After selecting the build, follow the steps to log in to your Apple Developer account. When prompted **Show devices and ask me again**, you can select a new provisioning profile.

> Select a new device, and the command will run the EAS Build again. Note that the build triggered this time is not a full build. It reuses the artifacts from the selected build and only codesign them to the new provisioning profile. Once the process of codesign is complete, you can use this new build link to install the **.ipa** on the iOS device whose provisioning profile was selected during the process.

Reviewer: ```suggestion
Select a new device, and the command will run the EAS Build again. Note that the build triggered this time reuses the application artifact from the selected build and codesigns it with the new provisioning profile. Once the process of codesigning is complete, you can use this new build link to install the **.ipa** on the iOS device that was added to the provisioning profile during the process.
```
